scipy/stats/tests/gen_studentized_range_mpmath_ref.py

```diff
@@ -0,0 +1,186 @@
+    CdfCase(1, 3, 10, general_atol, general_rtol),
+    CdfCase(10, 3, 10, general_atol, general_rtol),
+
+    CdfCase(0.1, 10, 10, general_atol, general_rtol),
+    CdfCase(1, 10, 10, general_atol, general_rtol),
+    CdfCase(10, 10, 10, general_atol, general_rtol),
+
+    CdfCase(0.1, 3, 100, general_atol, general_rtol),
+    CdfCase(1, 3, 100, general_atol, general_rtol),
+    CdfCase(10, 3, 100, general_atol, general_rtol),
+
+    CdfCase(0.1, 10, 100, general_atol, general_rtol),
+    CdfCase(1, 10, 100, general_atol, general_rtol),
+    CdfCase(10, 10, 100, general_atol, general_rtol),
+
+    CdfCase(1, 3, 1000, general_atol, general_rtol),
+    CdfCase(10, 3, 1000, 1e-10, general_rtol),
+]
+
+# PDF can use same cases as CDF
+pdf_cases = cdf_cases
+
+mom_atol, mom_rtol = 1e-9, 1e-9
+# These are EXTREMELY slow - Multiple days each in worst case.
+moment_cases = [
+    MomentCase(1, 3, 10, mom_atol, mom_rtol),
+    MomentCase(2, 3, 10, mom_atol, mom_rtol),
+    MomentCase(3, 3, 10, mom_atol, mom_rtol),
+    MomentCase(4, 3, 10, mom_atol, mom_rtol)
+]
+
+
+def to_dict(named_tuple):
+    return dict(named_tuple._asdict())
+
+
+def mp_res_to_dict(mp_result):
+    """Formats an MPResult namedtuple into a dict for JSON dumping"""
+    return {
+        "src_case": to_dict(mp_result.src_case),
+
+        # np assert can't handle mpf, so take the accuracy hit here.
+        "mp_result": float(mp_result.mp_result)
+    }
+
+
+def cdf_mp(q, k, nu):
+    """Calculates the CDF integrand for the given q, k, nu vals"""
+    q, k, nu = mpf(q), mpf(k), mpf(nu)
+
+    def inner(s, z):
+        return phi(z) * (Phi(z + q * s) - Phi(z)) ** (k - 1)
+
+    def outer(s, z):
+        return s ** (nu - 1) * phi(sqrt(nu) * s) * inner(s, z)
+
+    def whole(s, z):
+        return (sqrt(2 * pi) * k * nu ** (nu / 2) /
+                (gamma(nu / 2) * 2 ** (nu / 2 - 1)) * outer(s, z))
+
+    res = quad(whole, [0, inf], [-inf, inf],
+               method="gauss-legendre", maxdegree=10)
+    return res
+
+
+def pdf_mp(q, k, nu):
+    """Calculates the PDF integrand for the given q, k, nu vals"""
+    q, k, nu = mpf(q), mpf(k), mpf(nu)
+
+    def integral(s, z):
+        return (k * (k - 1) * s * phi(z) * phi(s * q + z)
+                * (Phi(s * q + z) - Phi(z)) ** (k - 2))
+
+    def whole(s, z):
+        return (nu ** (nu / 2) / (gamma(nu / 2) * 2 ** (nu / 2 - 1)) *
+                s ** (nu - 1) * exp(-nu * s ** 2 / 2) * integral(s, z))
+
+    res = quad(whole, [0, inf], [-inf, inf],
+               method="gauss-legendre", maxdegree=10)
+    return res
+
+
+def moment_mp(m, k, nu):
+    """Calculates the moment integrand for the given m, k, nu vals"""
+    m, k, nu = mpf(m), mpf(k), mpf(nu)
+
+    def integral(q, s, z):
+        return k * (k - 1) * s * phi(z) * phi(s * q + z) \
+               * (Phi(s * q + z) - Phi(z)) ** (k - 2)
+
+    def whole(q, s, z):
+        return q ** m * nu ** (nu / 2) / (
+                gamma(nu / 2) * 2 ** (nu / 2 - 1)) * s ** (
```

Comment:
Yeah, please run `flake8` on this and everything.

Reply:
I made the formatting nicer on this, but I don't get any `flake8` messages about this. Do you have a nonstandard `flake8` configuration?

Reply:
Huh. I think if it were not a multiple of the default indentation it would have complained; I guess I couldn't tell. (You did get other flake8 messages though, no?)

Reply:
> For that file, it complained about some lambda expressions. Are you seeing anything else about it?

No, I didn't run it, and that was the other thing my eyes noticed.